filebeat/input/v2/input.go

```diff
@@ -114,27 +120,50 @@ func (c *Context) UpdateStatus(status status.Status, msg string) {
 // the PipelineClientListener or a beat.CombinedClientListener. It's the latter
 // when clientListener is non-nil.
 // The PipelineClientListener collects pipeline metrics for an input. The
-// metrics are created on reg.
+// metrics are created on reg. If there is already a metric with the same name
+// on reg, the existing metric will be used.
 func NewPipelineClientListener(
 	reg *monitoring.Registry,
 	clientListener beat.ClientListener) beat.ClientListener {
 
-	var pcl beat.ClientListener = &PipelineClientListener{
-		eventsTotal: monitoring.NewUint(
-			reg, "events_pipeline_total"),
-		eventsFiltered: monitoring.NewUint(
-			reg, "events_pipeline_filtered_total"),
-		eventsPublished: monitoring.NewUint(
-			reg, "events_pipeline_published_total"),
+	pcl := &PipelineClientListener{}
+
+	if monVar := reg.Get(metricEventsPipelineTotal); monVar == nil {
+		pcl.eventsTotal = monitoring.NewUint(
+			reg, metricEventsPipelineTotal)
+	} else {
+		if monUInt, ok := monVar.(*monitoring.Uint); ok {
+			pcl.eventsTotal = monUInt
+		}
```

Comment:
This looks like it could be factored out in to a helper.

Reply:
done